staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/subresource.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fieldmanager
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal"
+	"sigs.k8s.io/structured-merge-diff/v4/fieldpath"
+)
+
+// FindEntries returns all the managed fields which has a specific path
+func FindEntries(entries []metav1.ManagedFieldsEntry, pathParts ...interface{}) ([]metav1.ManagedFieldsEntry, error) {
+	path, err := fieldpath.MakePath(pathParts...)
+	if err != nil {
+		return nil, err
+	}
+
+	var filteredEntries []metav1.ManagedFieldsEntry
+	for _, entry := range entries {
+		if entry.FieldsV1 == nil {
```

Comment:
It's kind of sad that you have to decide to decide that here. If we had more than one "Fields" version, we would have to remember to change that rather than using logic that probably/should exist somewhere else.

Reply:
The only thing that comes to mind to avoid this is to change `internal.FieldsToSet` to handle the case when its argument is `nil`. What do you think? Do you have something else in mind?

Reply:
I suspect you're not supposed to use this function, but `fieldmanager.DecodeObjectManagedFields` instead, since it transforms the whole `ManagedFields` into the proper type (and you don't have to consider each individual field). Let me know if the return type doesn't give you what you need.

Reply:
It's definitely possible to use `DecodeObjectManagedFields` to check whether an entry has a certain field, but I am having hard time to change the entry to a specific version or change the field set. Do you have any thoughts on that?